### PR TITLE
chore(kafka sink): add `BytesSent` metric

### PR DESCRIPTION
**Pull request by fuchsnj**
I reviewed the new-style sinks to make sure they are all sending `EventsSent` + `BytesSent` and have appropriate documentation for them. Only kafka ended up needing some changes.

As part of this, I also moved `BytesSent` into `core-common`.

### CODE REVIEW

File: lib/vector-core/core-common/src/internal_event/bytes_sent.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+use crate::internal_event::InternalEvent;
```
Comment (bruceg) [inline lib/vector-core/core-common/src/internal_event/bytes_sent.rs:1]: I just realized that the `check-events` scanner won't be looking at these. Do you want to add this to the script, or leave that to me in a separate issue?

**Review by bruceg — APPROVED** [review]
LGTM

Please also make sure #9325 is up to date with your review. I linked this PR into the list.

**Comment by bruceg** [comment]
Oh, I just realized this is missing component tests. Could you add those to this?

File: lib/vector-core/core-common/src/internal_event/bytes_sent.rs (hunk 1 above)
Comment (fuchsnj) [reply lib/vector-core/core-common/src/internal_event/bytes_sent.rs:1]: I added it.

**Pull request merged**